packages/widgetbook/lib/src/addons/device_addon/device_addon.dart

```diff
@@ -26,62 +26,73 @@ class DeviceAddon extends WidgetbookAddOn<DeviceSetting> {
         );
 
   @override
-  Widget buildSetting(BuildContext context) {
-    return Setting(
-      name: name,
-      child: Row(
-        children: [
-          DropdownSetting<Device?>(
-            options: setting.devices,
-            optionValueBuilder: (device) => device?.name ?? 'None',
-            initialSelection: setting.activeDevice,
-            onSelected: (device) {
-              updateSetting(
-                setting.copyWith(
-                  activeDevice: device,
-                ),
-              );
-            },
+  List<Field> get fields {
+    return [
+      ListField<Device?>(
+        group: slugName,
+        name: 'name',
+        values: setting.devices,
+        labelBuilder: (device) => device?.name ?? 'None',
+        codec: FieldCodec(
+          toParam: (device) => device?.name ?? 'None',
+          toValue: (param) => setting.devices.firstWhere(
+            (device) => device?.name == param,
+            orElse: () => null,
           ),
-          if (setting.activeDevice != null) ...{
-            IconButton(
-              tooltip: 'Orientation',
-              onPressed: () {
-                updateSetting(
-                  setting.copyWith(
-                    orientation: setting.orientation == Orientation.portrait
-                        ? Orientation.landscape
-                        : Orientation.portrait,
-                  ),
-                );
-              },
-              icon: Icon(
-                setting.orientation == Orientation.portrait
-                    ? Icons.screen_lock_portrait
-                    : Icons.screen_lock_landscape,
-                color: Theme.of(context).iconTheme.color,
-              ),
+        ),
+        onChanged: (device) {
+          if (device == null) return;
+
+          updateSetting(
+            setting.copyWith(
+              activeDevice: device,
+            ),
+          );
+        },
+      ),
+      ListField<Orientation>(
+        group: slugName,
+        name: 'orientation',
+        values: Orientation.values,
+        labelBuilder: (orientation) =>
+            orientation.name.substring(0, 1).toUpperCase() +
+            orientation.name.substring(1),
+        codec: FieldCodec(
+          toParam: (orientation) => orientation.name,
+          toValue: (param) => Orientation.values.byName(
+            param ?? Orientation.portrait.name,
+          ),
+        ),
+        onChanged: (orientation) {
+          if (orientation == null) return;
```

Comment:
How can this be null if the generic parameter is non-nullable?

Reply:
The `onChanged` here is not the same as the widget `onChanged`. It gets triggered in the following cases:
1. UI Change
2. Query Param Change: where this field can be null in that case if it does not exist in the query param.

Reply:
Ah, good to know 💡